inference-contract/docs/notebooks/templates/token.py

```diff
@@ -0,0 +1,182 @@
+token_owner = 'user1'
+token_class = 'mytoken'
+token_name = 'token_1'
+token_path = 'token.${token_class}.token_object.${token_name}'
+context_file = '${etc}/${token_class}_context.toml'
+service_host = 'localhost'
+instance_identifier = ''
+
+# %%
+import os
+import pdo.contracts.jupyter as pc_jupyter
+import IPython.display as ip_display
+
+pc_jupyter.load_ipython_extension(get_ipython())
+
+
+# %% [markdown]
+# ### Initialize the PDO Environment
+#
+# Initialize the PDO environment. This assumes that a functional PDO configuration
+# is in place and that the PDO virtual environment has been activated. In particular,
+# ensure that the groups file and eservice database have been configured correctly.
+This can be done most easily by running the following in a shell:
+#
+# `$PDO_HOME/bin/pdo-create-service-groups.psh --service_host <service_host>`
+#
+# For the most part, no modifications should be required below.
+
+# %%
+common_bindings = {
+    'host' : service_host,
+    'service_host' : service_host,
+    'token_owner' : token_owner,
+    'token_class' : token_class,
+    'token_name' : token_name,
+}
+
+(state, bindings) = pc_jupyter.initialize_environment(token_owner, **common_bindings)
+print('environment initialized')
+
+# %% [markdown]
+# ### Initialize the Contract Context
+#
+# The contract context defines the configuration for a collection of contract
+# objects that interact with one another. By default, the context file used in
+# this notebook is specific to the toke class used to mint the token.
+# We need the class to ensure that all of the information necessary for the token
+# itself is available. If you prefer to use a common context file, edit the
+# context_file variable below.
+#
+# For the most part, no other modifications should be required.
+
+# %%
+token_class_path = 'token.' + token_class
+context_file = bindings.expand(context_file)
+print("using context file {}".format(context_file))
+
+context = pc_jupyter.ml_jupyter.initialize_token_context(state, bindings, context_file, token_class_path)
+print('context initialized')
+
+# %% [markdown]
+# <hr style="border:2px solid gray">
+#
+# ## Operate on the Contract
+
+# %%
+token_context = pc_jupyter.pbuilder.Context(state, token_path)
+
+
+# %% [markdown]
+# ### Invoke Inference Operation
+#
+# The model is an image classfication model. Use the following cell to
+# provide input to the classification oepration. Current token object policy
+# simply checks ownership of the token. There are no additional policy checks
+# in this example.
+#
+# For the parameters to be provided below, if you are running the jupyter server
+# via the contracts docker container, copy your test image to
+# `$PDO_CONTRACTS_SRC_ROOT/docker/xfer/client/` folder and
+# provide the value `/project/pdo/xfer/client` for the input `path`.
+#
+# Also, please note that `$PDO_CONTRACTS_SRC_ROOT/inference-contract/data` folder
+# contains the sample image `zebra_wiki.jpg` that may be used for testing
```

Comment:
Can we mention here also where in container that zebra_wiki.jpg file is located (i.e., `/project/pdo/contracts/inference-contract/data`)?

Reply:
thanks, have re-written the block of text to clarify better.